scipy/interpolate/_ndgriddata.py

```diff
@@ -116,8 +122,16 @@ def __call__(self, *args):
         xi = _ndim_coords_from_arrays(args, ndim=self.points.shape[1])
         xi = self._check_call_shape(xi)
         xi = self._scale_x(xi)
-        dist, i = self.tree.query(xi)
-        return self.values[i]
+        dist, i = self.tree.query(xi, distance_upper_bound=self.query_max_dist)
```

Comment:
Since this new argument is forwarded to `kdtree.query`, maybe better keep the name consistent. Maybe even expose `**kwargs` and forward them all to `self.tree.query`. Not sure if they all make sense for the `NearestNDInterpolator`, so maybe start with `distance_upper_bound` and `workers`. Not sure about the rest, WDYT?

Reply:
Good catch on also exposing workers. The rest don't seem useful. Having **kwargs might be a little dangerous because the interpolator will break if k is not set to 1. I have exposed workers and distance_upper_bound (renamed) separately in the latest commit: https://github.com/scipy/scipy/pull/19483/commits/2fa01c9110a7c4db730625ac2efb57e94b8d7f3d. Let me know what you think